lwt/xen_api_lwt_unix.ml

```diff
@@ -62,12 +62,16 @@ module Lwt_unix_IO = struct
     Ssl.create_context Ssl.TLSv1_2 Ssl.Client_context
 
   let open_connection uri =
-    let domain_addr_t = match Uri.host uri with
+    let domain_addr_t port = match Uri.host uri with
       | Some host ->
         Lwt.catch
           (fun () ->
-             Lwt_unix.gethostbyname host >>= fun host_entry ->
-             return (host_entry.Lwt_unix.h_addrtype, host_entry.Lwt_unix.h_addr_list.(0)))
+             let host = Scanf.ksscanf host (fun _ _ -> host) "[%s@]" Fun.id in
+             Lwt_unix.getaddrinfo host (string_of_int port) [] >>= fun he ->
+             if he = [] then raise Not_found ;
+
+             let sockaddr = (List.hd he).Unix.ai_addr in
```

Comment:
let sockaddr = match he with
| [] -> raise Not_found
| addrinfo :: _ -> addrinfo.Unix.ai_addr
in